pkg/etcdtools/virtualization.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2016 Mirantis
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcdtools
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	etcd "github.com/coreos/etcd/client"
+	"golang.org/x/net/context"
+)
+
+type VirtualizationTool struct {
+	keysAPITool *KeysAPITool
+}
+
+func NewVirtualizationTool(keysAPITool *KeysAPITool) (*VirtualizationTool, error) {
+	kapi, err := keysAPITool.newKeysAPI()
+	if err != nil {
+		return nil, err
+	}
+	if _, err := kapi.Set(context.Background(), "/container", "", &etcd.SetOptions{Dir: true}); err != nil {
+		// 102 "Not a file error" means that the dir node already exists.
+		// There is no way to tell etcd client to ignore this fact.
+		// TODO(nhlfr): Report a bug in etcd about that.
+		if !strings.Contains(err.Error(), "102") {
+			return nil, err
+		}
+	}
```

Comment:
Hmmm... This block is repeated in https://github.com/Mirantis/virtlet/pull/16/commits/473c3778e1faf9fa6e391e0da0d8ac31f604795b#diff-3fa1b3884d9211562318b8a6453862e4R45 and https://github.com/Mirantis/virtlet/pull/16/commits/473c3778e1faf9fa6e391e0da0d8ac31f604795b#diff-562748e967d3c48749c184eab3f09410R414
This could be refactored and then this will be a single place to contain this comment.